File: spark-scala/spark-datasources/spark-hive/src/main/scala/com/mahfooz/spark/hive/metastore/SparkHiveMetastore.scala
/*

 */
package com.mahfooz.spark.hive.metastore

import org.apache.spark.sql.SparkSession

object SparkHiveMetastore {

  def main(args: Array[String]): Unit = {

    val spark=SparkSession.builder()
      .master("local")
      .appName("SparkHiveMetastore")
      .config("spark.sql.warehouse.dir","/user/hive/warehouse")
      .config("hive.metastore.uris","thrift://mtmdevhdopms01.intl.vsnl.co.in:9083")
      .enableHiveSupport()
      .getOrCreate()

    val dbs=spark.sqlContext.sql("show databases")

    dbs.show()
    spark.stop()
  }

}

File: spark-python/pyspark-sql/src/sql/metastore/hive/spark_hive_metastore.py
"""
Spark SQL uses a Hive metastore to manage the metastore of persistent relational entities (e.g.,
databases, tables, columns, partitions) in a relational database (for fast access).
A Hive metastore (aka metastore_db) is a relational database to manage the metastore of the
persistent relational entities, e.g., databases, tables, columns, partitions.
By default, Spark SQL uses the embedded deployment mode of a Hive metastore with an Apache Derby database.
"""
import os

from pyspark.sql import SparkSession

if __name__ == '__main__':
    warehouse_location = os.environ["SPARK_WAREHOUSE"]
    derby_home = os.environ["derby.system.home"]

    spark = (SparkSession.builder
             .appName("spark_hive_metastore")
             .config("spark.sql.catalogImplementation", "hive")
             .config("spark.driver.extraJavaOptions",
                     f"-Dderby.system.home='{derby_home}'")
             .enableHiveSupport()
             .getOrCreate())

    spark.sql("show tables in global_temp").show()